### DISCUSSION TITLE
UMEP outside QGIS

### DISCUSSION THREAD

**Discussion by kwazineutin** [UMEP-dev/UMEP#368] [Python]
Hi,

Is it possible to have a tutorial on how the functions in UMEP can be used outside the QGIS environment, for example in jupyter notebooks?

Thanks.

**Comment by biglimp** [comment]
The intention is to create this kind of tutorial but there are too many things in the pipline before this can happen. Here, I call our to the UMEP community to write such a tutorial.

There are actually information that can help you on the way. First have a look at on how to set up a Python environment for Windows [here](https://umep-docs.readthedocs.io/en/latest/DevelopmentGuidelines.html#quick-guide-to-set-up-a-python-ide-on-windows). Then you can go through [this tutorial](https://umep-docs.readthedocs.io/projects/tutorial/en/latest/Tutorials/IntrodutionToProcessingSEBE.html#processingsebe) and try to make a stand alone script. If you want to use Jyputer Notebook you can also install this in your OsGeo Python environment according to the instructions [here](https://umep-docs.readthedocs.io/en/latest/Getting_Started.html#adding-missing-python-libraries-and-other-osgeo-functionalities).

**Comment by j3r3m1** [comment]
Hi @kwazineutin ,
we have initialized 2 tutorials concerning your need to use Python for processing UMEP outside QGIS.
Can you have a try to the following tutorials and let us know your feedback and if it answers your need:

- https://umep-docs.readthedocs.io/projects/tutorial/en/latest/Tutorials/PythonProcessing1.html
- https://umep-docs.readthedocs.io/projects/tutorial/en/latest/Tutorials/PythonProcessing2.html

**Reply by kwazineutin** [reply]
hi @j3r3m1 These are very informative.
Thank you.

**Reply by j3r3m1** [reply]
Is it what you were expecting (does it answer your need) ?

**Reply by kwazineutin** [reply]
Yes, this meets my need at the moment.

Are there any plans of having UMEP as a conda package in future?

thanks.
